src/Lightning.ts

```diff
@@ -807,8 +807,24 @@ export const LightningMixin = (superclass) =>
         // TODO: we should not log/keep secret in the logs
         this.logger.debug({ invoice, user: this.user }, "got invoice status")
       } catch (err) {
-        const error = `issue fetching invoice`
-        this.logger.warn({ err, invoice }, error)
+        const invoiceNotFound = "unable to locate invoice"
+        try {
+          assert(err.length === 3 && err[2].err.details === invoiceNotFound)
+        } catch (err2) {
+          this.logger.error(
+            { err, err2, invoice },
+            "issue fetching invoice. unknown error",
+          )
+          return false
+        }
+
+        this.logger.info({ err, invoice }, invoiceNotFound)
+        try {
+          await InvoiceUser.deleteOne({ _id: hash, uid: this.user._id })
```

Comment:
Should this always be tried here?
I think this code block 811 - 827 could be structured clearer via explicit if/else control flow.